locallib.php

```diff
@@ -2033,8 +2033,12 @@ function bigbluebuttonbn_include_recording_table_row($bbbsession, $recording) {
     if (isset($recording['imported'])) {
         return true;
     }
-    // When groups are enabled, exclude those to which the user doesn't have access to.
-    if (isset($bbbsession['group']) && $recording['meetingID'] != $bbbsession['meetingid']) {
+    // When groups are enabled, exclude those to which the user doesn't have access to
+    // Unless we are administrator or moderator.
+    if (isset($bbbsession['group'])
```

Comment:
I guess this is doing what is supposed, but how could we reduce complexity? even at the price of "redundancy"

Reply:
Hi Jesus, 

I guess that I would try to push the conditional on group in the bigbluebuttonbn_get_recording_data_row, so to have only one location where the group is checked (the loop in bigbluebuttonbn_get_recording_table), so it is both used in bigbluebuttonbn_get_recording_data and bigbluebuttonbn_get_recording_table. But that seems to be hard to do at this point without disrupting another part of the code. 
My original question was there because I was not yet sure that the group condition was working for non admin/moderator users, I checked it after (it is in the ticket) and it works. So all is good.
So if we are happy with the actual code modification which is quite minimal, we can just keep this in mind when we will do a more thorough refactoring.